File: frontend/src/app/_data/Dataset.ts
import { FolderFile } from "./FolderFile";

export default class Dataset extends FolderFile {
    constructor(
        name: string = 'Novi izvor podataka',
        public description: string = '',
        public fileId?: string,
        public extension: string = '.csv',
        public isPublic: boolean = false,
        public accessibleByLink: boolean = false,
        dateCreated: Date = new Date(),
        lastUpdated: Date = new Date(),
        public uploaderId: string = '',
        public delimiter: string = ',',

        public columnInfo: ColumnInfo[] = [],
        public rowCount: number = 0,
        public nullRows: number = 0,
        public nullCols: number = 0,
        public isPreProcess : Boolean = false,
        public cMatrix: number[][] = []
    ) {
        super(name, dateCreated, lastUpdated);
    }
}

export class ColumnInfo {
    constructor(
        public columnName: string = '',
        public isNumber: boolean = false,
        public numNulls: number = 0,
        public uniqueValues: string[]=[],
        public uniqueValuesCount: number[]=[],
        public uniqueValuesPercent: number[]=[],
        public median: number=0,
        public mean: number=0,
        public min: number=0,
        public max: number=0,
        public q1: number=0,
        public q3: number=0,
    ) {
        /*if (isNumber)
            this.columnType = ColumnType.numerical;
        else 
            this.columnType = ColumnType.categorical;*/
    }

}

File: backend/api/api/Models/Dataset.cs
using System;
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace api.Models
{
	public class Dataset
	{
        public Dataset() { }
        public string uploaderId { get; set; }

        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]//mongo data type to .net
        public string _id { get; set; }

        public string description { get; set; }
        public string name { get; set; }
        public string fileId { get; set; }
        public string extension { get; set; }
        public bool isPublic { get; set; }
        public bool accessibleByLink { get; set; }
        public DateTime dateCreated { get; set; }
        public DateTime lastUpdated { get; set; }
        public string delimiter { get; set; }
        public ColumnInfo[] columnInfo { get; set; }
        public int rowCount { get; set; }
        public int nullCols { get; set; }
        public int nullRows { get; set; }
        public bool isPreProcess { get; set; }

        public float[][] cMatrix { get; set; }
    }
}

